shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -103,68 +70,281 @@ async fn topology_task_process(
+            operation: CassandraOperation::Error(error),
+            ..
+        })) = message.frame()
+        {
+            return error.message == "unconfigured table peers_v2";
+        }
+
+        false
+    }
+
+    fn into_nodes(mut response: Message, config_data_center: &str) -> Result<Vec<CassandraNode>> {
+        if let Some(Frame::Cassandra(frame)) = response.frame() {
+            match &mut frame.operation {
+                CassandraOperation::Result(CassandraResult::Rows {
+                    value: MessageValue::Rows(rows),
+                    ..
+                }) => rows
+                    .iter_mut()
+                    .filter(|row| {
+                        if let Some(MessageValue::Varchar(data_center)) = row.last() {
+                            data_center == config_data_center
+                        } else {
+                            false
+                        }
                     })
-                })
-                .collect(),
-            operation => Err(anyhow!(
-                "system.peers returned unexpected cassandra operation: {:?}",
-                operation
-            )),
+                    .map(|row| {
+                        if row.len() != 4 && row.len() != 5 {
+                            return Err(anyhow!("expected 4 or 5 columns but was {}", row.len()));
+                        }
+
+                        let _data_center = row.pop();
+
+                        let tokens = if let Some(MessageValue::List(list)) = row.pop() {
+                            list.into_iter()
+                                .map::<Result<String>, _>(|x| match x {
+                                    MessageValue::Varchar(a) => Ok(a),
+                                    _ => Err(anyhow!("tokens value not a varchar")),
+                                })
+                                .collect::<Result<Vec<String>>>()?
+                        } else {
+                            return Err(anyhow!("tokens not a list"));
+                        };
+
+                        let rack = if let Some(MessageValue::Varchar(value)) = row.pop() {
+                            value
+                        } else {
+                            return Err(anyhow!("rack not a varchar"));
+                        };
+
+                        let ip = if let Some(MessageValue::Inet(value)) = row.pop() {
+                            value
+                        } else {
+                            return Err(anyhow!("native_address not an inet"));
+                        };
+
+                        let port = if let Some(message_value) = row.pop() {
+                            if let MessageValue::Integer(value, _) = message_value {
+                                value
+                            } else {
+                                return Err(anyhow!("port is not an integer"));
+                            }
+                        } else {
+                            9042
```

Comment:
same here but comment about system.peers_v2 vs system.peers

Reply:
comment doesnt quite make sense directly copied.
How about something like: "this method supports both system.peers and system.peers_v2, system.peers does not have a field for the port so we fallback to the default port."

Reply:
Ah yep, MB.